sentry-core/src/main/java/io/sentry/core/Scope.java

```diff
@@ -62,6 +71,33 @@ public void setFingerprint(List<String> fingerprint) {
   }
 
   public void addBreadcrumb(@NotNull Breadcrumb breadcrumb) {
+    addBreadcrumb(breadcrumb, true);
+  }
+
+  void addBreadcrumb(@NotNull Breadcrumb breadcrumb, boolean executeBeforeBreadcrumb) {
+    if (executeBeforeBreadcrumb && beforeBreadcrumbCallback != null) {
+      try {
+        breadcrumb =
+            beforeBreadcrumbCallback.execute(breadcrumb, null); // TODO: whats about hint here?
```

Comment:
We should take hint also

Reply:
@bruno-garcia how? if the user has called `scope.addBreadcrumb(...)` where do we get it from?

Reply:
We'd need to add an overload that takes hint.